sdk/src/main/scala/com/horizen/SidechainHistory.scala

```diff
@@ -480,9 +482,13 @@ object SidechainHistory
                                       historyBlockValidators: Seq[HistoryBlockValidator],
                                       stakeEpochInfo: StakeConsensusEpochInfo) : Try[SidechainHistory] = Try {
 
-    if (historyStorage.isEmpty)
+    if (historyStorage.isEmpty) {
+      val minimalHash = getMinimalHashOptFromBlock(genesisBlock).getOrElse(throw new IllegalStateException("Genesis block without mainchain block references"))
+      val nonceEpochInfo = NonceConsensusEpochInfo(bigIntToConsensusNonce(minimalHash))
```

Comment:
why this hack defined here but not in `SidechainNodeViewHolder.genesisState` method as well as `stakeEpochInfo`?